mlflow/metrics/metric_definitions.py

```diff
@@ -33,6 +33,28 @@ def _validate_text_data(data, metric_name, column_name):
     return True
 
 
+def _validate_and_fix_text_tuple_data(data, metric_name, column_name):
+    """Validates that the data is a list of a tuple of strings and is non-empty"""
+    if data is None or len(data) == 0:
+        return False
+
+    for row, tup in enumerate(data):
+        if not isinstance(tup, tuple) or not all(isinstance(val, str) for val in tup):
+            if isinstance(tup, str):
+                # Single entry tuples get unpacked.
+                # So if the entry is a string, put them back into a tuple.
+                data[row] = (tup,)
```

Comment:
Is there a reason this is doing some data mutation instead of just validating the data? 🤔 can we make this a util function which doesn't do this? Let me know I am missing something